client/a8c-for-agencies/components/site-configurations-modal/index.tsx

```diff
@@ -69,6 +72,11 @@ export default function SiteConfigurationsModal( {
 
 	const onSubmit = ( event: React.FormEvent< HTMLFormElement > ) => {
 		event.preventDefault();
+		if ( ! siteId ) {
```

Comment:
We could consider relying here on already-calculated `isDevSite` value.

```suggestion
		if ( isDevSite ) {
```

Reply:
That was my intention, but this is dueTypescript.
Later in the code, we use `siteId` as a number, which now is an optional `siteId?: number`, which triggers the lint.

![image](https://github.com/user-attachments/assets/33b7b257-4b34-44b4-a50f-4cf57ad2e78e)

Reading this logically, the `isDevSite` as **false** guarantees that siteId is not empty. But IDE and lint does not go so deep in logic.

We can improve that when implement the form submitting.

Reply:
Alright! I see. That's a good point! Thanks for clarifying.

> We can improve that when implement the form submitting.

Maybe we could then (not in this PR) consider passing a different `onSubmit` handler conditionally - where the form itself is built, e.g.:

```
<form onSubmit={ isDevSite ? createDevSite : createSite }>
```